### PR TITLE
fix(common-utils): recover select-alias map from unparseable queries

### PR DESCRIPTION
**Pull request by alex-fedotyev**
The select-alias map went empty whenever a rendered query contained ClickHouse-specific SQL that node-sql-parser's Postgresql dialect can't parse. The clearest case is a sampling CTE: filter value distributions render `greatest(CAST(total / N AS UInt32), 1)`, and the `CAST(... AS UInt32)` cast makes the whole statement unparseable. When the parse threw, `chSqlToAliasMap` returned `{}`, which drops the `WITH` clauses that define a source's select aliases. Anything filtering on an aliased column (`ServiceName as service`, a `ResourceAttributes['x']` map alias) then failed with `Unknown identifier`.

## Summary

`chSqlToAliasMap` now falls back to parsing only the outer `SELECT` projection when the full statement doesn't parse. The alias map only needs the `expr AS alias` pairs in the projection, so stripping the CTEs and the `WHERE` clause (where the unparseable SQL usually lives) recovers the aliases.

- Split the existing AST-to-alias-map extraction into a small helper so the happy path and the fallback share it (no behavior change on inputs that already parsed).
- Added `extractOuterSelectProjection`, a paren- and quote-aware scan that returns the text between the top-level `SELECT` and `FROM`. Nested subqueries and CTE bodies are skipped because their keywords sit inside parentheses, and keywords inside string literals are ignored.
- On a parse failure, re-parse `SELECT <projection> FROM <table>` once before giving up. If that also fails the map is empty, exactly as before.

## Why

This makes `aliasWith` reliable for every consumer of the alias map (the search results table, histogram, alerts, the CLI), not just the simple-select sources that already parsed. It is the groundwork for fixing alias filters on the Event Patterns view.

## Test plan

- [x] `make ci-lint` (eslint + tsc) on `@hyperdx/common-utils`
- [x] `make ci-unit` on `@hyperdx/common-utils` (1329 tests pass)
- [x] new unit tests in `clickhouse.test.ts` for the sampling-CTE / `CAST` case, bracket aliases, expression aliases, JSON-path aliases, and the string-literal edge case. Each fails on `main` and passes here.
- [x] confirmed the existing happy-path alias tests still return identical maps

[as-cast: allow] the `as SQLParser.Select` assertion is carried over verbatim from the existing implementation; node-sql-parser's `astify` returns a union type and the line keeps its `eslint-disable` comment.

### CODE REVIEW

File: packages/common-utils/src/clickhouse/index.ts [hunk 1]
```diff
@@ -819,12 +819,152 @@ export function parameterizedQueryToSql({
...
+function extractOuterSelectProjection(sql: string): string | null {
+  let parenDepth = 0;
+  let inSingleQuote = false;
+  let inDoubleQuote = false;
+  let inBacktick = false;
+  let projectionStart = -1;
+
+  const isWordChar = (c: string | undefined) =>
+    c != null && /[A-Za-z0-9_]/.test(c);
+  const matchesKeywordAt = (index: number, keyword: string): boolean => {
```
[the remark this answers is not in the document]
Comment (alex-fedotyev) [reply packages/common-utils/src/clickhouse/index.ts:899]: Fixed in 7742d30b3: the scanner now skips `--` line comments and `/* */` block comments, with a regression test.

**Review by jordan-simonovski — APPROVED** [review]
